orchagent/twamporch.cpp

```diff
@@ -0,0 +1,1052 @@
+
+#define TWAMP_SESSION_TIMEOUT_MIN    1
+#define TWAMP_SESSION_TIMEOUT_MAX    10
+
+static map<string, sai_twamp_session_role_t> twamp_role_map =
+{
+    { "SENDER",       SAI_TWAMP_SESSION_ROLE_SENDER    },
+    { "REFLECTOR",    SAI_TWAMP_SESSION_ROLE_REFLECTOR }
+};
+
+static map<string, sai_twamp_mode_t> twamp_mode_map =
+{
+    { "FULL",     SAI_TWAMP_MODE_FULL    },
+    { "LIGHT",    SAI_TWAMP_MODE_LIGHT   }
+};
+
+static map<string, sai_twamp_timestamp_format_t> timestamp_format_map =
+{
+    { "NTP",    SAI_TWAMP_TIMESTAMP_FORMAT_NTP },
+    { "PTP",    SAI_TWAMP_TIMESTAMP_FORMAT_PTP }
+};
+
+static map<string, bool> session_admin_state_map =
+{
+    { "ENABLED",  true  },
+    { "DISABLED", false }
+};
+
+static map<string, bool> hw_lookup_map =
+{
+    { "TRUE",  true  },
+    { "FALSE", false }
+};
+
+/* Global variables */
+extern sai_object_id_t     gSwitchId;
+extern sai_object_id_t     gVirtualRouterId;
+extern sai_switch_api_t   *sai_switch_api;
+extern sai_twamp_api_t    *sai_twamp_api;
+extern CrmOrch            *gCrmOrch;
+
+const vector<sai_twamp_session_stat_t> twamp_session_stat_ids =
+{
+    SAI_TWAMP_SESSION_STAT_RX_PACKETS,
+    SAI_TWAMP_SESSION_STAT_RX_BYTE,
+    SAI_TWAMP_SESSION_STAT_TX_PACKETS,
+    SAI_TWAMP_SESSION_STAT_TX_BYTE,
+    SAI_TWAMP_SESSION_STAT_DROP_PACKETS,
+    SAI_TWAMP_SESSION_STAT_MAX_LATENCY,
+    SAI_TWAMP_SESSION_STAT_MIN_LATENCY,
+    SAI_TWAMP_SESSION_STAT_AVG_LATENCY,
+    SAI_TWAMP_SESSION_STAT_MAX_JITTER,
+    SAI_TWAMP_SESSION_STAT_MIN_JITTER,
+    SAI_TWAMP_SESSION_STAT_AVG_JITTER
+};
+
+
+
+TwampOrch::TwampOrch(TableConnector confDbConnector, TableConnector stateDbConnector, SwitchOrch *switchOrch, PortsOrch *portOrch, VRFOrch *vrfOrch) :
+        Orch(confDbConnector.first, confDbConnector.second),
+        m_stateDbTwampTable(stateDbConnector.first, stateDbConnector.second),
+        m_switchOrch(switchOrch),
+        m_portsOrch(portOrch),
+        m_vrfOrch(vrfOrch)
+{
+    /* Set entries count to 0 */
+    m_maxTwampSessionCount = m_twampSessionCount = 0;
+
+    /* Get the Maximum supported TWAMP sessions */
+    SWSS_LOG_INFO("Get the Maximum supported TWAMP sessions");
+    sai_attribute_t attr;
+    attr.id = SAI_SWITCH_ATTR_MAX_TWAMP_SESSION;
+    sai_status_t status = sai_switch_api->get_switch_attribute(gSwitchId, 1, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("Twamp session resource availability is not supported. Skipping ...");
+        return;
+    }
+    else
+    {
+        m_maxTwampSessionCount = attr.value.u32;
+    }
+
+    /* Set MAX entries to state DB */
+    if (m_maxTwampSessionCount)
+    {
+        vector<FieldValueTuple> fvTuple;
+        fvTuple.emplace_back("MAX_TWAMP_SESSION_COUNT", to_string(m_maxTwampSessionCount));
+        m_switchOrch->set_switch_capability(fvTuple);
+    }
+    else
+    {
+        SWSS_LOG_WARN("Twamp session resource availability is not supported. Skipping ...");
```

Comment:
Please change to NOTICE as this is expected to be not supported on some platforms

Reply:
Done.